src/client/utils/hooks/useExternalDataCollection.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  QuoteCartQuery,
+  useExternalInsuranceDataQuery,
+  InsuranceDataCollection,
+  ExternalInsuranceDataQuery,
+} from 'data/graphql'
+import { getDataCollectionId } from 'api/quoteCartQuerySelectors'
+
+type Options = {
+  onCompleted?: (data: InsuranceDataCollection) => void
+}
+
+export const useExternalDataCollection = (
+  quoteCartQueryData?: QuoteCartQuery,
+  { onCompleted }: Options = {},
+) => {
+  const variants = quoteCartQueryData?.quoteCart.bundle?.possibleVariations
+  const exposure = variants ? getInsuranceExposure(variants) : undefined
+  const dataCollectionId = getDataCollectionId(quoteCartQueryData) ?? null
+
+  const { data } = useExternalInsuranceDataQuery({
+    skip: !dataCollectionId,
+    variables: dataCollectionId ? { reference: dataCollectionId } : undefined,
+    onCompleted(newData) {
+      const matchingDataCollection = getDataCollection(newData, exposure)
+      if (matchingDataCollection) onCompleted?.(matchingDataCollection)
+    },
+  })
+
+  return data ? getDataCollection(data, exposure) : null
+}
+
+type QuoteCartQuoteBundleVariant = NonNullable<
+  QuoteCartQuery['quoteCart']['bundle']
+>['possibleVariations'][number]
+
+/**
+ * Only Car is supported
+ */
+export const getInsuranceExposure = (
```

Comment:
Nit: module name does not prepare me to see other exported things than `useExternalDataCollection` hook.  Can we rename or split?

Reply:
oh good point, those should not be exported